### PR TITLE
Updated handlers to avoid calling virtual methods in the destructor

### PR DESCRIPTION
**Pull request by clydestanfield**
This updates the StreamHandler to no longer call a virtual method from inside its destructor.

### CODE REVIEW

File: modules/c++/logging/source/StreamHandler.cpp [hunk 1]
```diff
@@ -26,26 +26,40 @@
...
     // write prologue to stream
     write(mFormatter->getPrologue());
 }
 
-logging::StreamHandler::StreamHandler(io::OutputStream* stream,
-                                      logging::LogLevel level) :
-    logging::Handler(level)
+StreamHandler::StreamHandler(io::OutputStream* stream,
+                             LogLevel level) :
+    Handler(level)
```
Comment (asylvest) [inline modules/c++/logging/source/StreamHandler.cpp]: Need to add `mClosed` to the initializer list here

File: modules/c++/logging/source/StreamHandler.cpp [hunk 2]
```diff
@@ -57,19 +71,30 @@ void logging::StreamHandler::setFormatter(logging::Formatter* formatter)
...
+
+void StreamHandler::closeImpl()
+{
+    if (!mClosed)
+    {
+        // end log with formatter injection
+        write(mFormatter->getEpilogue());
+
+        // delete formatter
+        Handler::close();
```
Comment (asylvest) [inline modules/c++/logging/source/StreamHandler.cpp]: But... this is still virtual

File: modules/c++/logging/source/StreamHandler.cpp (hunk 1 above)
Comment (clydestanfield) [inline modules/c++/logging/source/StreamHandler.cpp]: Done

**Comment by clydestanfield** [comment]
The comments have been addressed

**Pull request merged**